src/main/java/org/jenkinsci/plugins/workflow/support/steps/build/BuildInfoAction.java

```diff
@@ -0,0 +1,60 @@
+package org.jenkinsci.plugins.workflow.support.steps.build;
+
+import hudson.model.Job;
+import hudson.model.Run;
+import jenkins.model.Jenkins;
+import hudson.model.InvisibleAction;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Queue;
+import java.util.concurrent.ConcurrentLinkedQueue;
+
+
+public class BuildInfoAction extends InvisibleAction {
+    private final Queue<BuildInfo> buildInfos = new ConcurrentLinkedQueue<>();
+
+    BuildInfoAction() {
+    }
+
+    BuildInfoAction(String projectName, int buildNumber) {
+        buildInfos.add(new BuildInfo(projectName, buildNumber));
+    }
+
+    public List<Run<?, ?>> getChildBuilds() {
+        List<Run<?, ?>> builds = new ArrayList<>();
+
+        for (BuildInfo buildInfo : buildInfos) {
```

Comment:
Not sure if `ConcurrentLinkedQueue.size()` is fast enough though. Maybe it will hammer the performance